gnd/src/main/java/com/google/android/gnd/ui/home/featuredetails/FeatureDetailsViewModel.java

```diff
@@ -82,6 +93,25 @@ public FeatureDetailsViewModel(
                 featureMutations, observationMutations, (f, o) -> !f.isEmpty() && !o.isEmpty()));
   }
 
+  /** Returns true if the user is {@link Role#OWNER} or {@link Role#MANAGER} of the project. */
+  private boolean isActiveUserAuthorizedToModifyFeature(Feature feature) {
+    Role role = userRepository.getUserRole(feature.getProject());
+    return role == Role.OWNER || role == Role.MANAGER;
+  }
+
+  /**
+   * Returns true if the selected feature is of type {@link FeatureType#POINT} and the user has
+   * permissions to modify the feature.
+   */
+  private boolean shouldMoveMenuOptionBeVisible(Feature feature) {
+    return isActiveUserAuthorizedToModifyFeature(feature) && feature.isPoint();
+  }
+
+  /** Returns true if the user has permissions to modify the feature. */
+  private boolean shouldDeleteMenuOptionBeVisible(Feature feature) {
+    return isActiveUserAuthorizedToModifyFeature(feature);
```

Comment:
Nit: So far we've just been calling the active user `User` (w/o "active"). Perhaps better to do that here for consistency (or change throughout).